test_conformance/conversions/fplib.cpp

```diff
@@ -29,8 +29,8 @@ static uint32_t clz(uint64_t value)
 
     for( num_zeros = 0; num_zeros < (sizeof(uint64_t)*8); num_zeros++)
     {
-        if(0x8000000000000000 & (value << num_zeros))
-            break;
+        volatile uint64_t v = 0x8000000000000000ull & (value << num_zeros);
```

Comment:
Were both the volatile and the "ull" required to fix this? Looking back at the issue, I don't think my previous explanation of undefined behavior on integer overflow makes sense since conversions to unsigned should be well-defined. 
The change looks fine to me--I'm just trying to understand if the code was incorrect or if it's a compiler bug workaround.

Reply:
i think it should be gcc 9.2 compiler bug, i tested with gcc 5.1, it's ok.
```
C:\conversion>gcc t.c

C:\conversion>a.exe
f = -0x1.000000p+63
e:1593835520   s:40    clz:0 data:8000000000000000 sign:1
f2=-0x1.000000p+63

C:\conversion>gcc -O2 t.c

C:\conversion>a.exe
f = -0x1.000000p+63
e:1593835520   s:40    clz:0 data:8000000000000000 sign:1
f2=-0x1.000000p+63

C:\conversion>gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=D:/tdm/bin/../libexec/gcc/x86_64-w64-mingw32/5.1.0/lto-wrapper.exe
Target: x86_64-w64-mingw32
Configured with: ../../../src/gcc-5.1.0/configure --build=x86_64-w64-mingw32 --enable-targets=all --enable-languages=ada,c,c++,fortran,lto,objc,obj-c++ --enable-libgomp --enable-lto --enable-graphite --enable-cxx-flags=-DWINPTHREAD_STATIC --disable-build-with-cxx --disable-build-poststage1-with-cxx --enable-libstdcxx-debug --enable-threads=posix --enable-version-specific-runtime-libs --enable-fully-dynamic-string --enable-libstdcxx-threads --enable-libstdcxx-time --with-gnu-ld --disable-werror --disable-nls --disable-win32-registry --prefix=/mingw64tdm --with-local-prefix=/mingw64tdm --with-pkgversion=tdm64-1 --with-bugurl=http://tdm-gcc.tdragon.net/bugs
Thread model: posix
gcc version 5.1.0 (tdm64-1)

C:\conversion>
```